flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/output/BoundedStreamTask.java

```diff
@@ -116,7 +116,7 @@ protected void processInput(MailboxDefaultAction.Controller controller) throws E
     protected void cancelTask() {}
 
     @Override
-    protected void cleanup() throws Exception {
+    protected void cleanUpInternal() throws Exception {
```

Comment:
I only see two methods now: `cleanUp` and `cleanUpInternal`.
They are meant to implement "template method" pattern with a usual naming in Flink (`public xxx()` calling `protected xxxInternal()`).

As for naming differently, the difference is not semantic - they both do some cleanup; `cleanUpInternal` just allows subclasses to hook into it.

WDYT?

Reply:
Yes, I think you can leave as is. I just emphasized that we have two methods with similar names and it is not clear which method should be overridden in which case. I mean having the one main method `cleanUp` which use several abstract(or just protected methods) like `cleanUpOperator`, `cleanUpTimers`, etc. is pretty understandable but having one `cleanUp` inside of another `cleanUp` is not so understandable. But I agree that it is not easy to come up with a good name because it is indeed the cleanUp so let's leave it as `cleanUpInternal`